### PR TITLE
Initial MongoDB migration

### PR DESCRIPTION
**Pull request by wolny**
TODO:
- [x] migration script to populate `immuno-assay-metadata` (see p. 3 in https://github.com/hci-unihd/batchlib/issues/58)

WIP
CC @constantinpape @tischi: I'm opening this WIP PR early, so that you can look at the README and see if there is something missing in the schema of `immuno-assay-metadata`. Ignore the schema of `immuno-assay-analysis-results` for now, as I'd like to discuss it in a separate PR where we implement the `DatabaseJob`.

### CODE REVIEW

File: batchlib/mongo/README.md [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+use covid
+db.foo.insert( { x: 1, y: 1 } )
+```
+
+## Replication
+TBD
+
+## Schema description
+The following collections are present in the `covid` database:
+- `immuno-assay-metadata` -  a collection to store all metadata associated with the plates
```
[the remark this answers is not in the document]
Comment (constantinpape) [reply batchlib/mongo/README.md:91]: One more thing: do we need to completely "fix" the json schema in advance?
I.e. if we add the dict `external_results` (or so), do we need to fix all the field names in advance, or can these be added on demand?

[the remark this answers is not in the document]
Comment (wolny) [reply batchlib/mongo/README.md:91]: Thanks for the comments @constantinpape.

> We def. want to store the manual assessment of images Vibor did in order to compare to the automatic results. And eventually we might want to add data from other assays as well, e.g. serology assays (ELISA) or the neutrilisation assay. I think I would just add one more dict for this on the well level and on the image level

there is only `manual_assessment` on the well level atm; but sure: adding additional object on the well level to store the info from ELISA and neutralization essay can be done any time. MongoDB is schemaless, so it won't break anything

> do we need to completely "fix" the json schema in advance?

not at all, the schema can evolve over time and the attributes can be freely added or removed; that's why using something schema free like Mongo is a good choice for use, going relational (SQL) would be a nightmare :)

[the remark this answers is not in the document]
Comment (constantinpape) [reply batchlib/mongo/README.md:91]: Ok, great!
Then I think it's good as it is.
Eventually, we could discuss if we would rather have separate fields
```
well: {elisa_assay: '', neurtilisation_assy: '', ...} 
```
or a sub-dictionary:
```
well: {..., assays: {...}}
```
But given that mongo is schema less, no reason to decide it now.

**Comment by wolny** [comment]
The initial migration is ready and tested. I'll merge it directly, cause this PR is fairly independent from existing codebase. When we have the plates data at EMBL, I'll setup mongo DB there and run the migration.

I'll continue with the `DatabaseJob` (see: https://github.com/hci-unihd/batchlib/issues/58#issuecomment-620757677) and DB API for the plate viewer.

**Pull request merged**